cashctrl_ledger/tests/test_accounts.py

```diff
@@ -125,44 +121,64 @@ def test_modify_account_with_invalid_group_raise_error(self, engine):
                 "group": "/ABC",
             })
 
-    def test_mirror_accounts_with_root_category(self, engine):
-        """Test ensure new account categories are created before mirroring and root nodes
-        remain untouched, as CashCtrl allows creating accounts only with existing categories.
+    def test_mirror_accounts_updates_category_tree(self, engine):
+        """
+        Ensures that new categories are created and orphaned categories,
+        except root nodes, are deleted when mirroring accounts.
+        """
+        ACCOUNT_CSV = """
+            group,                 account, currency, tax_code, description
+            /Balance,                 9990,      EUR,         , Test EUR Bank Account
+            /Balance/Node,            9993,      EUR,         , Transitory Account EUR
+            /Balance/Node/Subnode,    9994,      CHF,         , Transitory Account CHF
         """
-        engine.restore(accounts=ACCOUNTS, settings=self.SETTINGS)
+        ACCOUNTS = pd.read_csv(StringIO(ACCOUNT_CSV), skipinitialspace=True)
+
         initial_accounts = engine.accounts.list()
-        expected = initial_accounts[~initial_accounts["group"].str.startswith("/Balance")]
         initial_categories = engine._client.list_categories("account", include_system=True)
-        categories_dict = initial_categories.set_index("path")["number"].to_dict()
+        initial_categories = initial_categories["path"].to_list()
+        expected_categories = ACCOUNTS["group"].to_list()
+        assert not set(expected_categories).issubset(initial_categories), (
+            "Expected categories already exists"
+        )
 
-        assert not initial_accounts[initial_accounts["group"].str.startswith("/Balance")].empty, (
-            "There are no remote accounts placed in /Balance node"
+        # Ensure target categories not present on remote are created when mirroring
+        engine.accounts.mirror(ACCOUNTS, delete=True)
+        accounts = engine.accounts.list()
+        categories = engine._client.list_categories("account", include_system=True)
+        categories = categories["path"].to_list()
+        assert not accounts[accounts["group"].str.startswith("/Balance")].empty, (
+            "Accounts with '/Balance' root category were not created"
+        )
```

Comment:
Checking whether all accounts were mirrored would be more convincing.
I propose a test along the line of:
```
expected =pd.concat(ACCOUNTS, accounts.query("account not in @ACCOUNTS["account"]), ignore_index=True)
assert_frame_equal(expected, accounts, ignore_row_order=True)
```


Instead of:
```
assert set(initial_categories) == set(categories), (
    "Some orphaned categories were not deleted"
)
```